artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/impl/CoreMessage.java

```diff
@@ -1271,4 +1271,14 @@ public Object getOwner() {
    public void setOwner(Object object) {
       this.owner = object;
    }
+
+   @Override
+   public String getStringBody() {
```

Comment:
I think you check CoreMessage.type before you do this 

That is... if (type != Message.TEXT_TYPE) return null;

Reply:
you would get wrong decoding, and that could give you some head aches here. Please check the type.

Reply:
Thanks for the review, @clebertsuconic. I've made the change you suggested.